### ISSUE TITLE
Testing time issue

### ISSUE THREAD

**Issue by Newtog** [AlexeyAB/darknet#813]
Hi @AlexeyAB  I have followed the instruction for training on custom data. I have trained the model using yolov3-tiny.cfg . Since I have only 1 GPU. After training when I am trying to test I am getting below error.

I have used below command and got error 

ubuntu@ip-172-30-6-205:~/darknet-master/build/darknet/x64$ ./darknet detector test data/obj.data yolo-obj.cfg yolo-obj_58728_608_final.weights test/00000299.jpg

Error
 21 Layer before convolutional layer must output image.: File exists
darknet: ./src/utils.c:210: error: Assertion `0' failed.
Aborted (core dumped)

Is there something I am missing or some error please suggest thanks in advance

**Comment by Newtog** [comment]
I have changed height and width to 800*600 as it is actual frame size during training I kept 603*603 apart from this I have changed class number and filter at two place

**Comment by Newtog** [comment]
[net]
# Testing
batch=1
subdivisions=1
# Training
# batch=64
# subdivisions=2
width=800
height=600
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=30
activation=linear



[yolo]
mask = 3,4,5
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=5
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=30
activation=linear

[yolo]
mask = 1,2,3
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=5
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

**Comment by AlexeyAB** [comment]
You should use width and height any value multiple of 32. 
`800x608` and `608x608`


https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> 2. After training - for detection:
> Increase network-resolution by set in your .cfg-file (height=608 and width=608) or (height=832 and width=832) or (any value multiple of 32) - this increases the precision and makes it possible to detect small objects: link

**Comment by AlexeyAB** [comment]
Use flag `-dont_show` on Amazon EC2

`./darknet detector test data/voc.data yolo-voc.cfg yolo-voc.weights -dont_show`

And look at the file `predictions.jpg`

**Comment by Newtog** [comment]
Thanks

**Comment by Newtog** [comment]
Which one is better Yolov3.cfg or yolov3-tiny.cfg are they equivalent since first one is giving poor precision should I go for yolov3.cfg I have five classes and I have annotated 1000 for each

**Comment by AlexeyAB** [comment]
yolov3-tiny.cfg - faster but lower precision.

**Comment by Newtog** [comment]
Is it better then yolov2.cfg or not in terms of precision

**Comment by AlexeyAB** [comment]
Is it better then yolov2.cfg

**Comment by Newtog** [comment]
Thanks
